### ISSUE TITLE
What does the "all connections are in use" error mean?

### ISSUE THREAD

**Issue by N1h1l1sT** [rstudio/plumber#814]
So, on a server I have an API using plumber.
The R code uses another R package to download and filter some dataset and return the data as JSON.
Things were okay for like a day, but any request on anything now returns:

<simpleError in url("{SomeURL}/R_data/AQE_metadata.RData"): all connections are in use>
or
<simpleError in gzfile(file): all connections are in use>

What's weird is that on the same server, I copied the same code on a different folder, changed the plumber port to a different port so both can be up simultaneously, and the 1st one still fails whilst the 2nd one works using the same exact web request on the same server using the same code.

If this is not a plumber issue, I apologise and please close and delete this thread. It just seemed to me like it is, so I thought I'd open an issue here.

**Comment by schloerke** [comment]
Do you see any errors on the plumber server? Or do you only see errors on the requesting server?

----------------------

Is this within a Docker file? If so, can you produce the error outside of the Docker file?

-------------------

Is the server that is making the request closing the url connections after it is done being used?

Ex:
```r
html <- local({
	conn <- url("http://www.google.com")
	on.exit({close(conn) }, add = TRUE)
	readLines(conn)
})
html
```

**Comment by N1h1l1sT** [comment]
The errors I described are on the Plumber server ("simpleError in ..."). I see them on the log. The API runs with `/opt/R/4.0.5/bin/Rscript Service.R > /var/log/RAPI.log`
On the requesting side, like when I open a website with the link: `http://localhost:5431/GetImportMeta` I get this response: `{"error":"500 - Internal server error"}`
I also have an endpoint with just "/" where I just return the text "It's alive" and that comes through without error.

This is not within Docker.

I'm not sure if the requests are being closed. So there is this R library called "openair" which has lots of functions that return different data.
So, when I call the `/GetImportMeta` endpoint on my API, I'm using the openair::importMeta() which returns a tibble and I return this tibble.
So simplest case would be

```
#* @get /GetImportMeta
function() {
  Result <- importMeta(
      source = "aurn",
      all = False
    )

  return(jsonlite::toJSON(Result, digits = NA))
}
```

So, this works okay for like a day, and then it returns those errors.

**Comment by meztez** [comment]
To investigate further, add an endpoint that return the result of
```r
showConnections(all = TRUE)
```

It should tell you where your connections are used.

**Comment by N1h1l1sT** [comment]
So, I did exactly that to see if connections increase, but the /ShowAllConnections endpoint s[its out this:
`[["stdin","terminal","r","text","opened","yes","no"],["stdout","terminal","w","text","opened","no","yes"],["stderr","terminal","w","text","opened","no","yes"]]`
no matter how many other endpoints I call.

That seems to indicate that connections don't increase.
But if that's the case, why do I get the "all connections are in use" in the first place?
I don't seem to understand.

**Comment by schloerke** [comment]
TIL about `showConnections(all = TRUE)`.

`closeAllConnections()` should only be used in extreme measures, or if you fork the R process. (_Yay for websockets to RStudio IDE._)

--------------------------

The `all connections are in use` problem has existed before in `openair`: https://github.com/davidcarslaw/openair/search?q=connection+is%3Aissue&type=issues
Other functions were partially fixed in https://github.com/davidcarslaw/openair/pull/187 and https://github.com/davidcarslaw/openair/pull/40 .  This did not make me hopeful. So I went digging.

I tried to reprex your code without `plumber` to produce the error. I could not produce the connection error with your example. However, there are other types of data sources (https://github.com/davidcarslaw/openair/blob/5c54bad958854dd8abcd461d37a65c627a360b04/R/getMeta.R#L79) and they handle connections differently. 


<details>
<summary><code>source = "aurn"</code>; many temp files ❌ , but closes connections ✅ </summary>

``` r
library(openair)
source <- "aurn"
# This should have three entries
showConnections(all = TRUE)
#>   description class            mode text   isopen   can read can write
#> 0 "stdin"     "terminal"       "r"  "text" "opened" "yes"    "no"     
#> 1 "stdout"    "terminal"       "w"  "text" "opened" "no"     "yes"    
#> 2 "stderr"    "terminal"       "w"  "text" "opened" "no"     "yes"    
#> 3 "output"    "textConnection" "wr" "text" "opened" "no"     "yes"
# This should be empty
dir(tempdir())
#> character(0)

try({
  for (i in 1:100) {
    if (i %% 10 == 0) print(i)
    # Import the data
    importMeta(source = source, all = TRUE)
  }
})
#> [1] 10
#> [1] 20
#> [1] 30
#> [1] 40
#> [1] 50
#> [1] 60
#> [1] 70
#> [1] 80
#> [1] 90
#> [1] 100

# This should have three entries
showConnections(all = TRUE)
#>   description class            mode text   isopen   can read can write
#> 0 "stdin"     "terminal"       "r"  "text" "opened" "yes"    "no"     
#> 1 "stdout"    "terminal"       "w"  "text" "opened" "no"     "yes"    
#> 2 "stderr"    "terminal"       "w"  "text" "opened" "no"     "yes"    
#> 3 "output"    "textConnection" "wr" "text" "opened" "no"     "yes"
# This should be empty
dir(tempdir())
#>   [1] "file104e1105cc498" "file104e11157ec74" "file104e1128c0156"
#>   [4] "file104e1154b5149" "file104e11619fe42" "file104e116628c48"
#>   [7] "file104e11709a9d1" "file104e1186f79c6" "file104e118e0da78"
#>  [10] "file104e11935b62d" "file104e11afbd8e5" "file104e11c91f47f"
#>  [13] "file104e11c93d6be" "file104e11ec466b6" "file104e11f009d6b"
#>  [16] "file104e11fe7d673" "file104e12184b841" "file104e122251d9c"
#>  [19] "file104e12364bc91" "file104e12450348c" "file104e125abbe9d"
#>  [22] "file104e12742e3a5" "file104e129742099" "file104e129d3e50c"
#>  [25] "file104e12a8fac1d" "file104e12a97e5be" "file104e12b0f3f59"
#>  [28] "file104e12c93b813" "file104e12c97679d" "file104e12db02862"
#>  [31] "file104e12dca4ba8" "file104e12df7629a" "file104e12e6f48bd"
#>  [34] "file104e130e19f17" "file104e1324acc80" "file104e132bafe47"
#>  [37] "file104e1350b35e9" "file104e13659f8b3" "file104e1384d9cba"
#>  [40] "file104e13a6edf8c" "file104e13b1dab6a" "file104e13c2d1a14"
#>  [43] "file104e13c32559e" "file104e13c59a8b"  "file104e13e6cb9bf"
#>  [46] "file104e13ea3c2b2" "file104e141769e60" "file104e1441e710b"
#>  [49] "file104e1445d0a46" "file104e144be13a5" "file104e144e8c00b"
#>  [52] "file104e145167e4c" "file104e147bdd303" "file104e1483aa64" 
#>  [55] "file104e14c6a8409" "file104e14cb8014b" "file104e14cd6cba5"
#>  [58] "file104e14f2560f5" "file104e151321045" "file104e151628b00"
#>  [61] "file104e15227090a" "file104e1532c8221" "file104e153eab4db"
#>  [64] "file104e15416ca9d" "file104e15607c0e7" "file104e156ceddc9"
#>  [67] "file104e15a6c32ca" "file104e15b18ac72" "file104e15ce1b538"
#>  [70] "file104e160278d8c" "file104e1615eebb3" "file104e165029210"
#>  [73] "file104e166409f2b" "file104e167f626a2" "file104e1684ffbbe"
#>  [76] "file104e16f6c9436" "file104e170334681" "file104e17083212" 
#>  [79] "file104e17143543e" "file104e173006f33" "file104e1757a9a26"
#>  [82] "file104e176080224" "file104e1799edc64" "file104e17a4b7e6" 
#>  [85] "file104e17b4fec89" "file104e17b6be427" "file104e17c4c606" 
#>  [88] "file104e17f688ec0" "file104e18138b4a"  "file104e182b515d" 
#>  [91] "file104e1bc38c3f"  "file104e1c10881c"  "file104e1c299a9"  
#>  [94] "file104e1c5cb03a"  "file104e1c9316f2"  "file104e1c983585" 
#>  [97] "file104e1cc77051"  "file104e1d37fd2e"  "file104e1da9e5f4" 
#> [100] "file104e1f136f30"
```

<sup>Created on 2021-07-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

</details>


<details>
<summary><code>source = "kcl"</code>; 👏 👏 ✅ 

</summary>

``` r
library(openair)
source <- "kcl"
# This should have three entries
showConnections(all = TRUE)
#>   description class            mode text   isopen   can read can write
#> 0 "stdin"     "terminal"       "r"  "text" "opened" "yes"    "no"     
#> 1 "stdout"    "terminal"       "w"  "text" "opened" "no"     "yes"    
#> 2 "stderr"    "terminal"       "w"  "text" "opened" "no"     "yes"    
#> 3 "output"    "textConnection" "wr" "text" "opened" "no"     "yes"
# This should be empty
dir(tempdir())
#> character(0)

try({
  for (i in 1:100) {
    if (i %% 10 == 0) print(i)
    # Import the data
    importMeta(source = source, all = TRUE)
  }
})
#> [1] 10
#> [1] 20
#> [1] 30
#> [1] 40
#> [1] 50
#> [1] 60
#> [1] 70
#> [1] 80
#> [1] 90
#> [1] 100

# This should have three entries
showConnections(all = TRUE)
#>   description class            mode text   isopen   can read can write
#> 0 "stdin"     "terminal"       "r"  "text" "opened" "yes"    "no"     
#> 1 "stdout"    "terminal"       "w"  "text" "opened" "no"     "yes"    
#> 2 "stderr"    "terminal"       "w"  "text" "opened" "no"     "yes"    
#> 3 "output"    "textConnection" "wr" "text" "opened" "no"     "yes"
# This should be empty
dir(tempdir())
#> character(0)
```

<sup>Created on 2021-07-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

</details>


<details>
<summary><code>source = "saqn"</code>; 💥 </summary>


``` r
library(openair)
source <- "saqn"
# This should have three entries
showConnections(all = TRUE)
#>   description class            mode text   isopen   can read can write
#> 0 "stdin"     "terminal"       "r"  "text" "opened" "yes"    "no"     
#> 1 "stdout"    "terminal"       "w"  "text" "opened" "no"     "yes"    
#> 2 "stderr"    "terminal"       "w"  "text" "opened" "no"     "yes"    
#> 3 "output"    "textConnection" "wr" "text" "opened" "no"     "yes"
# This should be empty
dir(tempdir())
#> character(0)

try({
  for (i in 1:200) {
    if (i %% 25 == 0) print(i)
    # Import the data
    importMeta(source = source, all = TRUE)
  }
})
#> [1] 25
#> [1] 50
#> [1] 75
#> [1] 100
#> [1] 125
#> Error in url("http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData") : 
#>   all connections are in use

# This should have three entries
showConnections(all = TRUE)
#>     description                                                                   
#> 0   "stdin"                                                                       
#> 1   "stdout"                                                                      
#> 2   "stderr"                                                                      
#> 3   "output"                                                                      
#> 4   "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 5   "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 6   "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 7   "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 8   "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 9   "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 10  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 11  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 12  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 13  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 14  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 15  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 16  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 17  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 18  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 19  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 20  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 21  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 22  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 23  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 24  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 25  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 26  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 27  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 28  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 29  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 30  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 31  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 32  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 33  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 34  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 35  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 36  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 37  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 38  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 39  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 40  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 41  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 42  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 43  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 44  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 45  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 46  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 47  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 48  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 49  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 50  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 51  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 52  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 53  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 54  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 55  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 56  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 57  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 58  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 59  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 60  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 61  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 62  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 63  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 64  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 65  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 66  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 67  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 68  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 69  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 70  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 71  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 72  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 73  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 74  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 75  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 76  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 77  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 78  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 79  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 80  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 81  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 82  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 83  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 84  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 85  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 86  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 87  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 88  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 89  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 90  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 91  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 92  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 93  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 94  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 95  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 96  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 97  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 98  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 99  "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 100 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 101 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 102 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 103 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 104 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 105 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 106 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 107 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 108 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 109 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 110 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 111 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 112 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 113 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 114 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 115 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 116 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 117 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 118 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 119 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 120 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 121 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 122 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 123 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 124 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 125 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 126 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#> 127 "gzcon(http://www.scottishairquality.scot/openair/R_data/SCOT_metadata.RData)"
#>     class            mode text     isopen   can read can write
#> 0   "terminal"       "r"  "text"   "opened" "yes"    "no"     
#> 1   "terminal"       "w"  "text"   "opened" "no"     "yes"    
#> 2   "terminal"       "w"  "text"   "opened" "no"     "yes"    
#> 3   "textConnection" "wr" "text"   "opened" "no"     "yes"    
#> 4   "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 5   "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 6   "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 7   "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 8   "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 9   "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 10  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 11  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 12  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 13  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 14  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 15  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 16  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 17  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 18  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 19  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 20  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 21  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 22  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 23  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 24  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 25  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 26  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 27  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 28  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 29  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 30  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 31  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 32  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 33  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 34  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 35  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 36  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 37  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 38  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 39  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 40  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 41  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 42  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 43  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 44  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 45  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 46  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 47  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 48  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 49  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 50  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 51  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 52  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 53  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 54  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 55  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 56  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 57  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 58  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 59  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 60  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 61  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 62  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 63  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 64  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 65  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 66  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 67  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 68  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 69  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 70  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 71  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 72  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 73  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 74  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 75  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 76  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 77  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 78  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 79  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 80  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 81  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 82  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 83  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 84  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 85  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 86  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 87  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 88  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 89  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 90  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 91  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 92  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 93  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 94  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 95  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 96  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 97  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 98  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 99  "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 100 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 101 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 102 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 103 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 104 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 105 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 106 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 107 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 108 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 109 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 110 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 111 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 112 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 113 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 114 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 115 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 116 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 117 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 118 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 119 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 120 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 121 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 122 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 123 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 124 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 125 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 126 "gzcon"          "rb" "binary" "closed" "yes"    "no"     
#> 127 "gzcon"          "rb" "binary" "closed" "yes"    "no"
# This should be empty
dir(tempdir())
#> character(0)
```

<sup>Created on 2021-07-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
</details>


So if you call `importMeta(source = "saqn")` or `importMeta(source = "saqd")`, your connections will run out after ~ 125 calls.

------------------------------------

tl/dr; `openair` needs to handle connections like it does for type `kcl`(https://github.com/davidcarslaw/openair/blob/5c54bad958854dd8abcd461d37a65c627a360b04/R/getMeta.R#L180-L183) or the user's R process will run out of connections after many calls to `importMeta()`.
